### PR TITLE
[Compute] Modeling ASG association with VMSS (Service support this hence updati…

### PR DESCRIPTION
**Pull request by anuchandy**
…ng swagger)

* Compute supports associating ASG with VMSS. Today customer has to use ARM template for this. This PR update the compute spec 2018-06-01 with this (that way swagger represents the service).

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-06-01/compute.json [hunk 1]
```diff
@@ -6508,6 +6508,13 @@
           },
           "description": "Specifies an array of references to backend address pools of application gateways. A scale set can reference backend address pools of multiple application gateways. Multiple scale sets cannot use the same application gateway."
         },
+        "applicationSecurityGroups": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SubResource"
+          },
+          "description": "Specifies an array of references to application security group."
```
Comment (dsgouda) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2018-06-01/compute.json]: Is "application security group" definition the same as `SubResource`?

**Review by dsgouda — COMMENTED** [review]
Looks fine apart from the comment

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-06-01/compute.json (hunk 1 above)
Comment (anuchandy) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-06-01/compute.json]: yes they are same, it's a reference using id. This is consistent with other references (e.g. `applicationGatewayBackendAddressPools`, `loadBalancerBackendAddressPools `) in the same model

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**